src/pages/Main/Main.js

```diff
@@ -1,9 +1,73 @@
-import React from 'react';
+import React, { Component } from 'react';
+import Item from '../Shopping/Item/Item';
+import './Main.scss';
+
+export default class Main extends Component {
+  constructor(props) {
+    super(props);
+    this.slideContainer = React.createRef();
+  }
+
+  state = {
+    slideIndex: 0,
+  };
+
+  slideNext() {
+    this.setState({ slideIndex: this.state.slideIndex + 1 }, () => {
+      this.slideContainer.current.style.transition = '  transform 2s';
+      this.slideContainer.current.style.transform = `translateX(-${
+        1184 * this.state.slideIndex
+      }px)`;
+    });
+  }
+
+  slidePrevious() {
+    this.setState({ slideIndex: this.state.slideIndex - 1 }, () => {
+      this.slideContainer.current.style.transition = ' transform 1s';
+      this.slideContainer.current.style.transform = `translateX(-${
+        1184 * this.state.slideIndex
+      }px)`;
+    });
+  }
```

Comment:
- 👍  👍  👍  너무 잘 해주셨는데, 메서드를 선언할 때 화살표 함수를 사용해주세요!
```suggestion
  slideNext = () => {
    this.setState({ slideIndex: this.state.slideIndex + 1 }, () => {
      this.slideContainer.current.style.transition = '  transform 2s';
      this.slideContainer.current.style.transform = `translateX(-${
        1184 * this.state.slideIndex
      }px)`;
    });
  }

  slidePrevious = () => {
    this.setState({ slideIndex: this.state.slideIndex - 1 }, () => {
      this.slideContainer.current.style.transition = ' transform 1s';
      this.slideContainer.current.style.transform = `translateX(-${
        1184 * this.state.slideIndex
      }px)`;
    });
  }
```
- 일반 함수 선언식의 경우, this 가 고정되지 않고 함수가 선언되는 환경에 따라서 동적으로 할당됩니다. 따라서 React 에서는 일반 함수식으로 메서드를 선언한 경우 추가적으로 해당 함수의 this 를 선언되어 있는 class 로 고정시키는 bind 과정을 거쳐야 합니다.
- 그에 비해 화살표 함수의 경우, this 가 함수가 선언된 환경의 this 로 고정이 됩니다. 따라서 추가적으로 bind 하는 과정이 필요 없습니다.

Reply:
네 감사합니다!